clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/TestVeniceAvroMapper.java

```diff
@@ -162,28 +166,91 @@ public void testMapWithQuotaNotExceeded(int numReducers, int taskId) throws IOEx
         eq(MRJobCounterHelper.WRITE_ACL_FAILURE_GROUP_COUNTER_NAME.getGroupName()),
         eq(MRJobCounterHelper.WRITE_ACL_FAILURE_GROUP_COUNTER_NAME.getCounterName()),
         anyLong());
-    // Expect reporter to record these counters due to no early termination
+
+    // Expect the output collect to collect output due to no early termination
+    verify(output, times(getNumberOfCollectorInvocationForFirstMapInvocation(numReducers, taskId)))
+        .collect(any(), any());
+  }
+
+  @Test
+  public void testMapWithCompressionMetricCollectionEnabled() throws IOException {
+    final String keyFieldValue = "key_field_value";
+    final String valueFieldValue = "value_field_value";
+    AvroWrapper<IndexedRecord> wrapper = getAvroWrapper(keyFieldValue, valueFieldValue);
+    OutputCollector<BytesWritable, BytesWritable> output = mock(OutputCollector.class);
+    Reporter mockReporter = createMockReporterWithCount(1L);
+
+    // No need of reducers to test these metrics
+    int numReducers = 0;
+    int taskId = 0;
+
+    VeniceAvroMapper mapper = getMapper(numReducers, taskId, mapperJobConfig -> {
+      mapperJobConfig.set(COMPRESSION_METRIC_COLLECTION_ENABLED, "true");
+      mapperJobConfig.set(ZSTD_DICTIONARY_CREATION_REQUIRED, "true");
+      // TODO need to figure out how to mock a real object's method or a static method to test this
+      mapperJobConfig.set(ZSTD_DICTIONARY_CREATION_SUCCESS, "false");
+    });
+    mapper.map(wrapper, NullWritable.get(), output, mockReporter);
+
     verify(mockReporter, times(1)).incrCounter(
         eq(MRJobCounterHelper.TOTAL_KEY_SIZE_GROUP_COUNTER_NAME.getGroupName()),
         eq(MRJobCounterHelper.TOTAL_KEY_SIZE_GROUP_COUNTER_NAME.getCounterName()),
-        anyLong());
+        eq(16L));
+
+    verify(mockReporter, times(1)).incrCounter(
+        eq(MRJobCounterHelper.TOTAL_UNCOMPRESSED_VALUE_SIZE_GROUP_COUNTER_NAME.getGroupName()),
+        eq(MRJobCounterHelper.TOTAL_UNCOMPRESSED_VALUE_SIZE_GROUP_COUNTER_NAME.getCounterName()),
+        eq(18L));
+
+    // No compression: so same as uncompressed value
     verify(mockReporter, times(1)).incrCounter(
         eq(MRJobCounterHelper.TOTAL_VALUE_SIZE_GROUP_COUNTER_NAME.getGroupName()),
         eq(MRJobCounterHelper.TOTAL_VALUE_SIZE_GROUP_COUNTER_NAME.getCounterName()),
+        eq(18L));
+
+    // compressionMetricCollectionEnabled is enabled, but Zstd dict is not mocked, so gzip
+    // will be incremented, but not zstd
+    verify(mockReporter, times(1)).incrCounter(
+        eq(MRJobCounterHelper.TOTAL_GZIP_COMPRESSED_VALUE_SIZE_GROUP_COUNTER_NAME.getGroupName()),
+        eq(MRJobCounterHelper.TOTAL_GZIP_COMPRESSED_VALUE_SIZE_GROUP_COUNTER_NAME.getCounterName()),
+        eq(34L));
+
+    verify(mockReporter, never()).incrCounter(
+        eq(MRJobCounterHelper.TOTAL_ZSTD_COMPRESSED_VALUE_SIZE_GROUP_COUNTER_NAME.getGroupName()),
+        eq(MRJobCounterHelper.TOTAL_ZSTD_COMPRESSED_VALUE_SIZE_GROUP_COUNTER_NAME.getCounterName()),
```

Comment:
In Gzip, we have specified an actual data size, can we do the same for zstd?

Reply:
Added it here as well as some more tests around this. Didn't add before as I was trying to mock some functions of actual mapper object which was calling the original functions rather than whencalled() mocks, so now I ended up creating a derived class `TestVeniceAvroMapperClass.java` and override the function in question.